packages/preprocess/src/process-queries.cjs

```diff
@@ -12,31 +12,56 @@ const createDefaultProps = function (filename, componentDevelopmentMode, duckdbQ
 		const valid_ids = Object.keys(duckdbQueries).filter((queryId) =>
 			queryId.match('^([a-zA-Z_$][a-zA-Z0-9d_$]*)$')
 		);
-		queryDeclarations += `
-            import debounce from 'debounce';
-            import { browser } from '$app/environment';
-			import {profile} from '@evidence-dev/component-utilities/profile';
-			
-			// partially bypasses weird reactivity stuff with \`select\` elements
-			function data_update(data) {
-				${valid_ids.map((id) => `${id} = data.${id} ?? [];`).join('\n')}
-			}
-
-			$: data_update(data);
-
+		// queryDeclarations += `
+		//     import debounce from 'debounce';
+		//     import { browser } from '$app/environment';
+		// 	import {profile} from '@evidence-dev/component-utilities/profile';
+		//
+		// 	// partially bypasses weird reactivity stuff with \`select\` elements
+		// 	function data_update(data) {
+		// 		${valid_ids.map((id) => `${id} = data.${id} ?? [];`).join('\n')}
+		// 	}
+		//
+		// 	$: data_update(data);
+		//
+		//
+		//     ${valid_ids
+		// 					.map(
+		// 						(id) => `
+		//         let ${id} = data.${id} ?? [];
+		//         const _query_${id} = browser
+		// 			  ? debounce((query) => profile(__db.query, query).then((value) => ${id} = value), 200)
+		// 			  : (query) => (${id} = profile(__db.query, query, "${id}"));
+		//         $: _query_${id}(\`${duckdbQueries[id].replaceAll('`', '\\`')}\`);
+		//     `
+		// 					)
+		// 					.join('\n')}
+		// `;
+
+		const queryStores = valid_ids.map(
+			(id) => `
+		const _${id} = new QueryStore(
+			\`${duckdbQueries[id].replaceAll('`', '\\`')}\`,
+			queryFunc,
+			\`${id}\`,
+			{ initialData: profile(__db.query, \`${duckdbQueries[id].replaceAll('`', '\\`')}\`) }
```

Comment:
Shouldn't this be `data.${id} ?? profile(__db.query...)` to take advantage of query prerendering?